### PR TITLE
Bug 1826252: preserve integers decoding raw JSON values

### PR DESCRIPTION
**Pull request by p0lyn0mial**

### CODE REVIEW

File: vendor/k8s.io/kubernetes/Godeps/Godeps.json [hunk 1]
```diff
@@ -1462,6 +1558,34 @@
 			"ImportPath": "honnef.co/go/tools",
 			"Rev": "v0.0.1-2019.2.2"
 		},
+		{
+			"ImportPath": "k8s.io/api",
+			"Rev": "v0.18.0-beta.2"
```
Comment (p0lyn0mial) [inline vendor/k8s.io/kubernetes/Godeps/Godeps.json:1563]: does it look good?

Comment (sttts) [reply vendor/k8s.io/kubernetes/Godeps/Godeps.json:1563]: @marun we should do another rebase to final version as soon as possible. Basically same procedure, only that picking into kube fork will be trivial, and tests most probably stay green.

Comment (sttts) [reply vendor/k8s.io/kubernetes/Godeps/Godeps.json:1563]: rc.1. Why beta.2?

File: vendor/k8s.io/kube-openapi/pkg/util/proto/validation/types.go [hunk 2]
```diff
@@ -210,7 +210,7 @@ func (item *primitiveItem) VisitPrimitive(schema *proto.Primitive) {
 		}
 	case proto.Number:
 		switch item.Kind {
-		case proto.Number:
+		case proto.Integer, proto.Number:
```
Comment (sttts) [inline vendor/k8s.io/kube-openapi/pkg/util/proto/validation/types.go:213]: this commit should have the glide.yaml change too

File: vendor/k8s.io/kubernetes/Godeps/Godeps.json (hunk 1 above)
Comment (sttts) [reply vendor/k8s.io/kubernetes/Godeps/Godeps.json:1563]: ok, for whatever reason (you discussed with @marun if I have understood correctly) all are beta.2 here. So it fine for api as well.

File: vendor/k8s.io/kube-openapi/pkg/util/proto/validation/types.go (hunk 2 above)
Comment (p0lyn0mial) [reply vendor/k8s.io/kube-openapi/pkg/util/proto/validation/types.go:213]: I think it does 

```
{
"ImportPath": "k8s.io/kube-openapi",
"Rev": "61e04a5be9a6"
},
```

Link: https://github.com/openshift/origin/blob/487752ab83a9c39deb6b3b6292aceb94f2e89f0c/vendor/k8s.io/kubernetes/Godeps/Godeps.json#L1606

Comment (sttts) [reply vendor/k8s.io/kube-openapi/pkg/util/proto/validation/types.go:213]: Does github hide it here? I only see the types.go change here in this commit.

Comment (sttts) [reply vendor/k8s.io/kube-openapi/pkg/util/proto/validation/types.go:213]: or do we completely rely on glide reading Godeps? Then it's fine.

**Pull request merged**